src/main/kotlin/notification/application/NotificationSearcher.kt

```diff
@@ -0,0 +1,20 @@
+package notification.application
+
+import note.domain.Note
+import notification.domain.Notification
+import notification.domain.NotificationRepository
+import shared.domain.Identifier
+
+class NotificationSearcher(
+    private val repository: NotificationRepository,
+) {
+    /***
+     * Looks for all Notifications with the given Note identifier in the repository.
+     * @see Notification
+     * @see Note
+     * @return A collection of Notification instances (empty if none found)
+     */
+    fun search(noteId: String) = search(Identifier(noteId))
```

Comment:
Creating an `Identifier` from an invalid string throws `InvalidUUIDException`.
The `KDoc` should relfect that.